openid/src/main/java/fish/payara/security/openid/OpenIdUtil.java

```diff
@@ -82,32 +78,19 @@ public static String readConfiguredValueFromMetadataOrProvider(String metadataVa
 
     public static Set<String> readConfiguredValueFromMetadataOrProvider(String[] metadataValue, JsonObject providerDocument, String openIdConstant, Config provider, String openIdProviderMetadataName) {
         Set<String> value;
-        // PayaraConfig can contain strings from microprofile config, e.g. parse set with '|' as separator.
         if (metadataValue.length == 0 && providerDocument.containsKey(openIdConstant)) {
-            value = parseSet(getConfiguredValue(String.class, null, provider, openIdProviderMetadataName));
-            if (value == null) {
+            String[] valueArr = getConfiguredValue(String[].class, metadataValue, provider, openIdProviderMetadataName);
+            if (valueArr == null) {
```

Comment:
`valueArr` is never null. it will be equal to `metadataValue`. You likely want to retain that `null` in second argument

Reply:
fixed